src/annotations/AnnotationManager.php

```diff
@@ -299,6 +306,22 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if ($this->_traitsSupported && $member_name) {
```

Comment:
You're right. I haven't looked at original code fragment for a while. Code for reading class annotations even if method/property annotations were requested was in there long time before `@stop` annotation was even introduced. Probably it did something special at a time, but it doesn't now.

And since you pointed out, that we're not doing annotation query from all methods/properties at once then there is no need to do it globally on class annotation reading.